packages/core/src/fhirpath/atoms.ts

```diff
@@ -54,6 +54,9 @@ export class SymbolAtom implements Atom {
     if (this.name === '$this') {
       return input;
     }
+    if (this.name[0] === '%') {
+      return [context.variables[this.name.slice(1)]];
```

Comment:
open to requiring the names to be exact e.g variables = {'%previous': {}, '%current': {}} vs. slicing

Reply:
1. If we add a new `VariableAtom`, this would be handled there.
2. I like the slicing approach, but I'm also open to other opinions
3. What should we do on `undefined variable`?  Throw?